### PR TITLE
Export api urls yaml

### PR DESCRIPTION
**Pull request by andreia-oca**
# Pull Request Template

## Type of change

-   [x] 🍕 New feature
-   [ ] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [ ] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Description

This PR allows one single deploy command for fullstack projects with functions on the backend and a frontend.

To allow this, the user will be able to expand and inject environment variables at frontend build time in the `genezio.yaml`.

For both `genezio deploy [--frontend]` and `genezio local`, the following trick can be used in the `scripts` field:
```
    build:
      - ls -la
      - VITE_HELLO_WORLD_FUNCTION_URL=${{functionHelloWorldApiUrl}} npm run build
    start:
      - npm install --silent
      - VITE_HELLO_WORLD_FUNCTION_URL=${{functionHelloWorldApiUrl}} npm run dev --silent
```

The variables of format `${{functionCamelCaseNameApiUrl}}` will expand to the function URL - `https://<uuid>.<region>.cloud.genez.io`.

For `genezio local` the function URL will expand to `http://localhost:8083/.functions/function-hello`.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [x] I have added tests;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/utils/scripts.ts [hunk 1]
```diff
@@ -4,6 +4,67 @@ import colors from "colors";
...
+    const expandedScripts: string[] | string = [];
+
+    for (let script of scripts) {
+        const regex = /\$\{\{[a-zA-Z0-9-]+\}\}/g;
+        const matches = script.match(regex);
+        const prefix = "${{";
+        const suffix = "}}";
+
+        if (matches) {
+            for (const match of matches) {
```
Comment (costinsin) [inline src/utils/scripts.ts:35]: I would avoid extra indentation

```suggestion
        for (const match of matches || []) {
```

File: src/utils/scripts.ts [hunk 2]
```diff
@@ -4,6 +4,67 @@ import colors from "colors";
 import _ from "lodash";
 import { Logger } from "tslog";
 import { execaCommand } from "execa";
+import { debugLogger, log } from "./logging.js";
+
+export async function expandVariablesFromScript(
```
Comment (costinsin) [inline src/utils/scripts.ts:9]: Looks like this function expands only function URLs variables. I would name it to something more self explanatory. Maybe: `expandFunctionURLVariablesFromScript`

File: tests/expandVariablesFromScript.test.ts
Comment (costinsin) [inline tests/expandVariablesFromScript.test.ts:1]: Thanks for writing tests! It's much appreciated!

File: src/commands/deploy/genezio.ts [hunk 3]
```diff
@@ -675,7 +676,26 @@ export async function deployFrontend(
 
     try {
         await doAdaptiveLogAction(`Building frontend ${index + 1}`, async () => {
-            await runScript(frontend.scripts?.build, frontend.path);
+            debugLogger.debug(
+                `Running build script for frontend ${index + 1}`,
+                frontend.scripts?.build,
+            );
```
Comment (costinsin) [inline src/commands/deploy/genezio.ts:682]: Is this relevant to be printed before expansion? I would print the script content after expansion.

File: src/commands/deploy/genezio.ts [hunk 4]
```diff
@@ -675,7 +676,26 @@ export async function deployFrontend(
...
+
+            const projectEnvDetails = await getProjectEnvFromProjectByName(name, stage);
+            if (!projectEnvDetails) {
+                throw new UserError("Project environment not found.");
+            }
+
+            const functions = projectEnvDetails.functions?.map((f) => ({
+                name: f.name.replace(/-([a-z])/g, (g) => g[1].toUpperCase()) + "ApiUrl",
```
Comment (costinsin) [inline src/commands/deploy/genezio.ts:690]: Why to uppercase and not to lowercase? It means that I should write variables like this: `${{HELLOApiUrl}}`. You are also doing this in local.

**Review by costinsin — APPROVED** [review]
LGTM

File: src/commands/deploy/genezio.ts (hunk 4 above)
Comment (andreia-oca) [reply src/commands/deploy/genezio.ts:687]: Nope, it won't be capitalized, it will be camelCase: `${{helloApiUrl}}`.

I extracted the logic in a utils method `kebabToCamelCase`. This should increase readability.

File: src/utils/scripts.ts (hunk 1 above)
Comment (andreia-oca) [reply src/utils/scripts.ts:35]: Agree. Done

File: tests/expandVariablesFromScript.test.ts
[the remark this answers is not in the document]
Comment (andreia-oca) [reply tests/expandVariablesFromScript.test.ts:1]: I am not sure what you mean.

File: src/utils/scripts.ts (hunk 2 above)
Comment (andreia-oca) [reply src/utils/scripts.ts]: Done

File: src/commands/deploy/genezio.ts (hunk 3 above)
Comment (andreia-oca) [reply src/commands/deploy/genezio.ts]: Nope, I forgot it here, thanks!
